src/migrator/utils.ts

```diff
@@ -158,3 +145,14 @@ WHERE "migration"."model name" = ${1}`,
 
 	return JSON.parse(data.executed_migrations) as string[];
 };
+
+export const migrationTablesExist = async (tx: Tx) => {
+	return (
+		await Promise.all(
+			['migration', 'migration lock'].map(async (tableName) => {
+				const result = await tx.tableList(`name = '${tableName}'`);
+				return result.rows.length > 0;
+			}),
+		)
+	).reduce((prevCheck, currCheck) => prevCheck && currCheck, true);
```

Comment:
It'd make much more sense to use `every` here:
```suggestion
	).every((check) => check);
```
It's also got the benefit that it'll short-circuit return as necessary (although on only two items that doesn't matter, but in general it's nice)

Reply:
Thx, very elegant!